src/gui/transferlistsortmodel.cpp

```diff
@@ -215,12 +253,14 @@ int TransferListSortModel::compare(const QModelIndex &left, const QModelIndex &r
 
     case TransferListModel::TR_DLLIMIT:
     case TransferListModel::TR_DLSPEED:
-    case TransferListModel::TR_PRIVATE:
     case TransferListModel::TR_QUEUE_POSITION:
     case TransferListModel::TR_UPLIMIT:
     case TransferListModel::TR_UPSPEED:
         return customCompare(leftValue.toInt(), rightValue.toInt());
 
+    case TransferListModel::TR_PRIVATE:
+        return customCompare(leftValue, rightValue);
```

Comment:
In addition to https://github.com/qbittorrent/qBittorrent/pull/20951/files#r1676758486
It seems you really need unique behavior here so I would suggest you create unique comparison for it and don't use general ones.
Something like `int privateFlagCompare(QVariant, QVariant)`.

Reply:
Done.